pkg/types/intoto/v0.0.1/entry.go

```diff
@@ -73,20 +73,42 @@ func NewEntry() types.EntryImpl {
 func (v V001Entry) IndexKeys() ([]string, error) {
 	var result []string
 
-	h := sha256.Sum256([]byte(v.env.Payload))
-	payloadKey := "sha256:" + hex.EncodeToString(h[:])
-	result = append(result, payloadKey)
+	// add digest over entire DSSE envelope
+	if v.IntotoObj.Content != nil && v.IntotoObj.Content.Hash != nil {
+		hashkey := strings.ToLower(fmt.Sprintf("%s:%s", swag.StringValue(v.IntotoObj.Content.Hash.Algorithm), swag.StringValue(v.IntotoObj.Content.Hash.Value)))
+		result = append(result, hashkey)
+	} else {
+		log.Logger.Error("could not find content digest to include in index keys")
+	}
+
+	// add digest over public key
+	if v.IntotoObj.PublicKey != nil && v.keyObj != nil {
+		key := *v.IntotoObj.PublicKey
```

Comment:
Use `CanonicalValue()`?

To match https://github.com/sigstore/rekor/blob/408b752c60f67e0efca3e8965360dfe28f957cdf/pkg/api/index.go#L63-L68